contracts/modules/cover/Cover.sol

```diff
@@ -286,6 +280,37 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     emit CoverEdited(coverId, params.productId, segmentId, msg.sender, params.ipfsData);
   }
 
+  function expireCover(uint coverId) external {
+    uint segmentId = _coverSegments[coverId].length - 1;
+    CoverSegment memory lastSegment = coverSegmentWithRemainingAmount(coverId, segmentId);
+    CoverData memory cover = _coverData[coverId];
+    uint expiration = lastSegment.start + lastSegment.period;
+
+    // require last segment to be expired
+    if (expiration > block.timestamp) {
+      revert CoverNotYetExpired(coverId);
+    }
+    for (uint j = 0; j < coverSegmentAllocations[coverId][segmentId].length; j++) {
+      PoolAllocation memory allocation =  coverSegmentAllocations[coverId][segmentId][j];
+      AllocationRequest memory allocationRequest;
+      // editing just the needed props for deallocation
+      allocationRequest.productId = cover.productId;
+      allocationRequest.allocationId = allocation.allocationId;
+      allocationRequest.previousStart = lastSegment.start;
+      allocationRequest.previousExpiration = expiration;
+
+      stakingPool(allocation.poolId).requestAllocation(
+        0, // amount
+        0, // previous premium
+        allocationRequest
+      );
+
+    }
+    // remove cover amount from from expiration buckets
+    uint bucketAtExpiry = Math.divCeil(expiration, BUCKET_SIZE);
+    activeCoverExpirationBuckets[cover.coverAsset][bucketAtExpiry] -= lastSegment.amount;
```

Comment:
We also need to check if the bucket was not processed already otherwise we'd be subtracting twice.